crates/dbsp/src/trace/spine_async/mod.rs

```diff
@@ -187,15 +459,27 @@ impl MergeStats {
 
 /// Persistence optimized [trace][crate::trace::Trace] implementation based on
 /// collection and merging immutable batches of updates.
+///
+/// This spine works asynchronously.  The batches exposed to cursors are
+/// maintained separately from the batches currently being merged by an
+/// asynchronous thread. When one or more merges complete, the spine fetches the
+/// new (smaller) collection of batches from the thread in the next step. (It
+/// could fetch them earlier, but it might be unfriendly to expose potentially
+/// one form of data to a given cursor and then a different form to the next one
+/// within a single step.)
 #[derive(SizeOf)]
 pub struct Spine<B>
 where
     B: Batch,
 {
     #[size_of(skip)]
     factories: B::Factories,
-    /// `levels` holds `MAX_LEVEL` number of BTrees which contain batches.
-    levels: Vec<BTreeMap<BatchIdent, BatchState<B>>>,
+
+    /// All the batches in the spine, in no particular order.
+    ///
+    /// Each batch must be non-empty.
+    batches: Vec<Arc<B>>,
```

Comment:
they aren't ordered by size anymore? @ryzhyk at some point mentioned this might be important for some operations (for perf) but I don't remember the details

Reply:
@ryzhyk Can you comment on this? These `batches` are the ones that go into the `CursorList`. Does the order matter?

Reply:
In the original implementation of the spine, they had to be ordered by size for efficient merging so we can always merge similar-size batches when possible. I haven't read this PR yet, but I assume it has some other way to optimize merge order. I don't think that the ordering of batched in CursorList makes a big difference.

Reply:
Yes, the ordering of the batches here is decoupled from the order in the merger, so probably this is OK here.

Reply:
Why do we need this variable here instead of just reading the latest list of batches from the merger on-demand? E.g., in `cursor()` it might be preferable to use the latest batches, since they merger may have made some progress optimizing them.